terraform/imagebuild/main.tf

```
@@ -56,11 +56,11 @@ resource "null_resource" "build_and_push_sample_apps" {
   count = length(local.sample_apps_dockerfile_list)
 
   provisioner "local-exec" {
-    command = "docker build -t ${data.aws_ecr_repository.sample_app.repository_url}:${dirname(element(local.sample_apps_dockerfile_list, count.index))}-latest ../../sample-apps/${dirname(element(local.sample_apps_dockerfile_list, count.index))}/"
-  }
-
-  provisioner "local-exec" {
-    command = "docker push ${data.aws_ecr_repository.sample_app.repository_url}:${dirname(element(local.sample_apps_dockerfile_list, count.index))}-latest"
+    command = <<-EOT
+      docker buildx create --use --name sapbuilder.${count.index} --platform=linux/arm64,linux/amd64 --driver  docker-container --bootstrap
```

```suggestion
      docker buildx create --use --name sapbuilder.${count.index} --platform=linux/arm64,linux/amd64 --driver  docker-container --bootstrap
```
What is the reason for having to create a new builder instance for each sample app? Is it because these builds are done in parallel and thus we require a builder for each one?

---

Yes. that's correct. Had to create distinct builders for this reason.